sklearn/model_selection/_split.py

```diff
@@ -274,13 +274,38 @@ class _BaseKFold(BaseCrossValidator, metaclass=ABCMeta):
     """Base class for KFold, GroupKFold, and StratifiedKFold"""
 
     @abstractmethod
-    def __init__(self, n_splits, *, shuffle, random_state):
+    def __init__(
+        self,
+        n_splits,
+        *,
+        shuffle,
+        random_state,
+        x_shape=None,
+        max_train_size=None,
+        test_size=None,
+    ):
+        if isinstance(n_splits, int):
+            # self.n_splits = n_splits
+            n_splits = int(n_splits)
+
         if not isinstance(n_splits, numbers.Integral):
-            raise ValueError(
-                "The number of folds must be of Integral type. "
-                "%s of type %s was passed." % (n_splits, type(n_splits))
-            )
-        n_splits = int(n_splits)
+            if (
```

Comment:
Since `n_splits` being a `str` would be valid only for `TimeSeriesSplit`, it seems that we should make the computation of `n_splits` in the specialized class and just pass the validate `n_splits` to the parent class.